server/game/GameActions/DiscardAtRandom.js

```diff
@@ -8,7 +8,7 @@ class DiscardAtRandom extends GameAction {
     }
 
     canChangeGameState({ amount = 1, player }) {
-        return amount >= 1 && player.hand.length >= 1;
+        return amount >= 1 && player.hand.length >= 1 && !player.cannotBeDiscardedAtRandom;
```

Comment:
As mentioned in other comment, I'd remove this in favor of ensuring `GameAction.allow` is working properly for the resolved action in Heads on Spikes. :)